src/wallet/SignMessageMiddleware.ts

```diff
@@ -17,12 +17,20 @@ export function makeSignMessageMiddleware(walletThunk: () => ethers.Wallet) {
           const address = await wallet.getAddress()
           // @ts-expect-error todo: parse params
           assert.equal(address, ethers.utils.getAddress(req.params[1]))
+
           // @ts-expect-error todo: parse params
```

Comment:
I am not sure that I need to add a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) instead.